tensorflow/core/platform/cpu_info.cc

```diff
@@ -344,5 +344,27 @@ int CPUModelNum() {
 #endif
 }
 
+int CPUIDNumSMT() {
+#ifdef PLATFORM_IS_X86
+  // https://software.intel.com/en-us/articles/intel-64-architecture-processor-topology-enumeration
+  // https://software.intel.com/en-us/articles/intel-sdm (Vol 3A)
```

Comment:
Adding "section Detecting Hardware Multi-threading Support and Topology" after Vol 3A would be helpful.